src/rest/APIRequest.js

```diff
@@ -31,6 +31,10 @@ class APIRequest {
   }
 
   make() {
+    if (!agent) {
+      agent = new https.Agent({ ...this.client.options.http.agent, keepAlive: true });
+    }
```

Comment:
Since we're now requiring v16, let's use my favourite feature!
```suggestion
    agent ??= new https.Agent({ ...this.client.options.http.agent, keepAlive: true });
```